pkgs/development/tools/misc/autogen/default.nix

```diff
@@ -28,6 +28,13 @@ stdenv.mkDerivation rec {
       pname = "31_allow_overriding_AGexe_for_crossbuild";
       sha256 = "0h9wkc9bqb509knh8mymi43hg6n6sxg2lixvjlchcx7z0j7p8xkf";
     })
+    # Next upstream release will contain guile-3 support. We apply non-invasive
```

Comment:
Where did you see that the next release will support guile 3?
I couldn't find such a commit in the upstream repo.
If there is one, we should fetch it from upstream instead of gentoo IMO.

Reply:
I think https://git.savannah.gnu.org/cgit/autogen.git/commit/?id=132576313e66ee270ebb70f82afdaeacec1f2fc1 adds guile-3 support (changes 39 files). The patch is not easy to backport as is.

https://sourceforge.net/p/autogen/bugs/196/ hints at autogen-master being able to work with guile-3 as is.

Unfortunately I was not able to generate working release tarball out of master tree to check if it has working state at all. master it requires working autogen to do generate rest if it's bits (and something magic on top I could not make to work locally). Failing at that I asked for a new at https://savannah.gnu.org/support/?110504 a few months ago.

Reply:
Ah, not easy indeed. I think you did the right thing then.